core/lib/Cargo.toml

```diff
@@ -37,6 +37,7 @@ base64 = "0.11"
 base16 = "0.2"
 pear = "0.1"
 atty = "0.2"
+httpdate = "0.3"
```

Comment:
`hyper` should already have all the date parsing we need, e.g. in <https://docs.rs/headers/0.3.1/headers/struct.IfModifiedSince.html>

Reply:
Do you know how to parse a `&str` into `IfModifiedSince`? I could not find a way a of creating an instance of `IfModifiedSince` (other headers implement `FromStr`). Additionally, the master branch depends on an old version of hyper so these features might not be available at all.

Reply:
In hyper 0.10 I believe we would want to go through [`Header::parse_header`](https://docs.rs/hyper/0.10.16/hyper/header/trait.Header.html#tymethod.parse_header), and for hyper 0.13 there is [`Header::decode`](https://docs.rs/headers/0.3.1/headers/trait.Header.html#tymethod.decode).

Reply:
Parsing works with `Header::parse_header`. However, the current version of hyper use time 0.1.x (with very limited API) and if I'm using time in `named_file.rs` I'm using time 0.2.x which makes it hard to write nice code when comparing the dates. Do you have any advice?

Reply:
I didn't realize it had a dependency on `time 0.1` for that... that means the difference in our code for hyper 0.10 and 0.13 is more than I expected, which is disappointing. I had been thinking of going ahead and applying this to the `master` branch and then updating it when we merge `async`, but that might be too complex of an approach if we are running into issues like these.

Reply:
I think in this case I should wait until you finished #1065